src/shims/time.rs

```diff
@@ -127,7 +190,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         // QueryPerformanceCounter uses a hardware counter as its basis.
         // Miri will emulate a counter with a resolution of 1 nanosecond.
-        let duration = Instant::now().duration_since(this.machine.time_anchor);
+        let duration = this.machine.threads.time_anchor.get();
```

Comment:
There's a `check_no_isolation` above that seems wrong now.
Why did the tests not catch that?